### ISSUE TITLE
Contradicting the docs, `--keep-unit` is required with `--register=no` to avoid using dbus

### ISSUE THREAD

**Issue by snarkmaster** [systemd/systemd#7228]
### Submission type

 - Bug report

### systemd version the issue has been seen with 

trunk

### Used distribution

any

### In case of bug report: Unexpected behaviour you saw & how to reproduce

This seems related to #6347.

I am trying to use `systemd-nspawn` without requiring dbus registration. To do this, I grabbed a CentOS7 system without a running `dbus`, and ran:

```
mkdir -p kitten/usr
cp /sbin/busybox kitten/usr
systemd-nspawn --register=no --directory=kitten /usr/busybox echo KITTEN
```

However, this failed with the error:

```
Failed to open system bus: No such file or directory
```

Looking at the code, it seems like the only way to get that while passing `--register=no` is to hit `allocate_scope()` here: https://github.com/systemd/systemd/blob/master/src/nspawn/nspawn.c#L3468

So the issue is that `allocate_scope` requires a working `dbus` connection. To avoid following the branch that calls `allocate_scope`, one has to additionally pass `--keep-unit`. So this works:

```
systemd-nspawn --register=no --keep-unit --directory=kitten /usr/busybox echo KITTEN
```

This behavior contradicts the documentation, which claims that `--keep-unit` has no effect when `--register=no` is passed.

> `--keep-unit` ... This has no effect if --register=no is used.  ...

**Comment by poettering** [comment]
indeed!

**Comment by poettering** [comment]
Fix waiting in #7364
